src/Microsoft.Extensions.Logging.AzureAppServices/Internal/BlobAppendReferenceWrapper.cs

```diff
@@ -1,46 +1,99 @@
+using System.Net;
+using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.WindowsAzure.Storage;
-using Microsoft.WindowsAzure.Storage.Blob;
 
 namespace Microsoft.Extensions.Logging.AzureAppServices.Internal
 {
     /// <inheritdoc />
     public class BlobAppendReferenceWrapper : ICloudAppendBlob
     {
-        private readonly CloudAppendBlob _cloudAppendBlob;
+        private readonly Uri _fullUri;
+        private readonly HttpClient _client;
+        private readonly Uri _appendUri;
 
-        /// <summary>
-        /// Creates new instance of <see cref="BlobAppendReferenceWrapper"/>.
-        /// </summary>
-        /// <param name="cloudAppendBlob">The <see cref="CloudAppendBlob"/> instance to wrap.</param>
-        public BlobAppendReferenceWrapper(CloudAppendBlob cloudAppendBlob)
+        public BlobAppendReferenceWrapper(string containerUrl, string name, HttpClient client)
         {
-            _cloudAppendBlob = cloudAppendBlob;
+            var uriBuilder = new UriBuilder(containerUrl);
+            uriBuilder.Path += "/" + name;
+            _fullUri = uriBuilder.Uri;
+
+            AppendBlockQuery(uriBuilder);
+            _appendUri = uriBuilder.Uri;
+            _client = client;
         }
+
         /// <inheritdoc />
-        public async Task<Stream> OpenWriteAsync(CancellationToken cancellationToken)
+        public async Task AppendAsync(ArraySegment<byte> data, CancellationToken cancellationToken)
+        {
+            async Task<HttpResponseMessage> AppendDataAsync()
+            {
+                var message = new HttpRequestMessage(HttpMethod.Put, _appendUri)
+                {
+                    Content = new ByteArrayContent(data.Array, data.Offset, data.Count)
+                };
+                AddCommonHeaders(message);
+
+                return await _client.SendAsync(message, cancellationToken);
+            }
+
+            var response = await AppendDataAsync();
+
+            if (response.StatusCode == HttpStatusCode.NotFound)
+            {
+                // If no blob exists try creating it
+                var message = new HttpRequestMessage(HttpMethod.Put, _fullUri)
+                {
+                    // Set Content-Length to 0 to create "Append Blob"
+                    Content = new ByteArrayContent(Array.Empty<byte>()),
+                    Headers =
+                    {
+                        { "If-None-Match", "*" }
+                    }
+                };
+
+                AddCommonHeaders(message);
+
+                var createResponse =
+                    await _client.SendAsync(message, cancellationToken);
+
+                // If result is not 2** or 412 throw, we don't know what to do with it
+                if (!createResponse.IsSuccessStatusCode &&
+                    createResponse.StatusCode != HttpStatusCode.PreconditionFailed)
+                {
+                    createResponse.EnsureSuccessStatusCode();
```

Comment:
This EnsureSuccessStatusCode is redundant, invert the if and call AppendDataAsync inside, otherwise fall through to the EnsureSuccessStatusCode call below.